### PR TITLE
Old client ringbuffer Filter problem

### PR DESCRIPTION
**Pull request by pveentjer**
Filter was not passed correctly

Tests have been added for various filter usages

### CODE REVIEW

**Comment by jerrinot** [comment]
could you add the test into the `ParallelTest` category?

**Comment by pveentjer** [comment]
Yes. But not as part of this PR.

File: hazelcast/src/main/java/com/hazelcast/client/ClientEngine.java [hunk 1]
```diff
@@ -59,6 +60,13 @@
     SecurityContext getSecurityContext();
 
     /**
+     * Returns the SerializationService.
+     *
+     * @return the SerializationService
+     */
+    SerializationService getSerializationService();
```
Comment (jerrinot) [inline hazelcast/src/main/java/com/hazelcast/client/ClientEngine.java]: is RingBuffer the only data structure which needs to expose `SerializationService`?

**Comment by pveentjer** [comment]
Taking too much time. Merging.

**Pull request merged**